graylog2-server/src/main/java/org/graylog2/plugin/streams/Stream.java

```diff
@@ -50,8 +50,6 @@ public static MatchingType valueOfOrDefault(String name) {
 
     String getContentPack();
 
-    Collection<AlertCondition> getAlertConditions();
```

Comment:
The method, as it was before, was plain broken. It worked only because of type erasure, as it was actually returning a `Collection<BasicDBObject>`. It was only used implicitly by Jackson when serializing a `Stream` instance as JSON, so it was also using a different representation that any other place that was serializing an `AlertCondition` instance and it was only identical by accident.

It was also a duplicate of `StreamService#getAlertConditions` which goes the hard path of instantiating an actual `AlertCondition`. Unfortunately we cannot just move that logic to the Stream implementation class (or pass the list of correct alert conditions to its constructor) as the `AlertCondition` instance itself [references the `Stream`](https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog2/plugin/alarms/AlertCondition.java#L35) which holds it.

It's not the cleanroom technique to just remove it for a minor version, but it's actually balancing between api breakage and offering a method that is just plain broken.